zipkin-storage/elasticsearch/src/test/java/zipkin2/elasticsearch/HttpCallTest.java

```diff
@@ -120,16 +134,58 @@
     call.clone().execute();
   }
 
-  @Test public void executionException_httpFailure() throws Exception {
+  @Test public void executionException_5xx() throws Exception {
     MOCK_RESPONSE.set(AggregatedHttpResponse.of(HttpStatus.INTERNAL_SERVER_ERROR));
 
-    Call<?> call = http.newCall(REQUEST, b -> null);
+    Call<?> call = http.newCall(REQUEST, BodyConverters.NULL);
 
     try {
       call.execute();
-      failBecauseExceptionWasNotThrown(IllegalStateException.class);
-    } catch (IllegalStateException expected) {
-      assertThat(expected).isInstanceOf(IllegalStateException.class);
+      failBecauseExceptionWasNotThrown(RuntimeException.class);
+    } catch (RuntimeException expected) {
+      assertThat(expected).hasMessage("response for / failed: 500 Internal Server Error");
+    }
+  }
+
+  @Test public void executionException_404() throws Exception {
+    MOCK_RESPONSE.set(AggregatedHttpResponse.of(HttpStatus.NOT_FOUND));
+
+    Call<?> call = http.newCall(REQUEST, BodyConverters.NULL);
+
+    try {
+      call.execute();
+      failBecauseExceptionWasNotThrown(FileNotFoundException.class);
```

Comment:
`assertThatThrownBy` more readable

Reply:
let's do this in another change because we use this pattern maybe a hundred different places.